src/modules/report/yaml.py

```diff
@@ -2,41 +2,51 @@
 
 from ruamel.yaml import YAML
 
-from collector import services, vulnerabilities
+from collector import services, vulnerabilities, services_lock, vulnerabilities_lock
 
 
 class YAMLReporter(object):
     def get_report(self):
         yaml = YAML()
+        vulnerabilities_lock.acquire()
```

Comment:
You don't need this lock here as you're not using the vulnerabilities variable here. 

(And you will go on to grab the lock inside get_vulnerabilities, which is the right place to do it as it's where you're accessing the variable you're protecting)